File: APCS202506_轉盤得分.cpp
#include <bits/stdc++.h>

using namespace std;

#define N 50010
typedef long long ll;

int main()
{
    // freopen("P_7_12_3.in", "r", stdin);
    int m, n, k, dir;
    cin >> m >> n >> k;
    vector<string> mat(m);
    for (int i = 0; i < m; ++i)
        cin >> mat[i];
    int tot = 0;
    while (k--)
    {
        for (int idx = 0; idx < m; ++idx)
        {
            cin >> dir;
            dir %= n;
            if (dir > 0)
                mat[idx] = mat[idx].substr(n - dir, dir) + mat[idx].substr(0, n - dir);
            else if (dir < 0)
            {
                dir = -dir;
                mat[idx] = mat[idx].substr(dir, n - dir) + mat[idx].substr(0, dir);
            }
        }
        for (int i = 0; i < n; ++i)
        {
            vector<int> cnt(26, 0);
            for (int j = 0; j < m; ++j)
                cnt[mat[j][i] - 'a']++;
            tot += *max_element(cnt.begin(), cnt.end());
        }
    }
    cout << tot;
    return 0;
}

File: APCS202506_轉盤得分.py
m,n,k = map(int,input().split())
mat = [ input() for _ in range(m)]
tot = 0
while k > 0:
    d = [int(x)%n for x in input().split()]
    for i,idx in enumerate(d):
        mat[i] = mat[i][-idx:]+mat[i][:-idx]
    for i in range(n):
        cnt=[0]*26
        for j in range(m):
            cnt[ord(mat[j][i])-ord('a')] += 1
        tot += max(cnt)
    k -= 1
print(tot)
